pkg/controllers/user/alert/deployer/notification_template.go

```diff
@@ -1,229 +1,254 @@
-{{ if (index .Alerts 0).Labels.namespace}}Namespace: {{(index .Alerts 0).Labels.namespace}}{{ end}}
-Expression: {{(index .Alerts 0).Labels.expression}}<br>
-Description: Threshold Crossed: datapoint value {{ (index .Alerts 0).Labels.current_value}} was {{ (index .Alerts 0).Labels.comparison}} to the threshold ({{ (index .Alerts 0).Labels.threshold_value}}) for ({{ (index .Alerts 0).Labels.duration}})<br>
-{{ end}}
-{{ if (index .Alerts 0).Labels.logs}}
-Logs: {{ (index .Alerts 0).Labels.logs}}
-{{ end}}
-{{ end}}
+	NotificationTmpl = `
+{{- define "rancher.title" -}}
+{{- if eq .CommonLabels.alert_type "event" }}
+{{ .CommonLabels.event_type}} event of {{.GroupLabels.resource_kind}} occurred
+
+{{- else if eq .CommonLabels.alert_type "systemService" }}
+The system component {{ .GroupLabels.component_name}} is not running
+
+{{- else if eq .CommonLabels.alert_type "nodeHealthy" }}
+The kubelet on the node {{ .GroupLabels.node_name}} is not healthy
+
+{{- else if eq .CommonLabels.alert_type "nodeCPU" }}
+The CPU usage on the node {{ .GroupLabels.node_name}} is over {{ .CommonLabels.cpu_threshold}}%
+
+{{- else if eq .CommonLabels.alert_type "nodeMemory" }}
+The memory usage on the node {{ .GroupLabels.node_name}} is over {{ .CommonLabels.mem_threshold}}%
+
+{{- else if eq .CommonLabels.alert_type "podNotScheduled" }}
+The Pod {{ if .GroupLabels.namespace}}{{.GroupLabels.namespace}}:{{end}}{{.GroupLabels.pod_name}} is not scheduled
+
+{{- else if eq .CommonLabels.alert_type "podNotRunning" }}
+The Pod {{ if .GroupLabels.namespace}}{{.GroupLabels.namespace}}:{{end}}{{.GroupLabels.pod_name}} is not running
+
+{{- else if eq .CommonLabels.alert_type "podRestarts" }}
+The Pod {{ if .GroupLabels.namespace}}{{.GroupLabels.namespace}}:{{end}}{{.GroupLabels.pod_name}} restarts {{ .GroupLabels.restart_times}} times in {{ .GroupLabels.restart_interval}} sec
+
+{{- else if eq .CommonLabels.alert_type "workload" }}
+The workload {{ if .GroupLabels.workload_namespace}}{{.GroupLabels.workload_namespace}}:{{end}}{{.GroupLabels.workload_name}} has available replicas less than {{ .GroupLabels.available_percentage}}%
+
+{{- else if eq .CommonLabels.alert_type "metric" }}
+The metric {{ .CommonLabels.alert_name}} crossed the threshold 
+{{ end -}}
+{{ end -}}
+
+{{- define "wechat.text" -}}
+{{ template "__wechat_text_list" . }}
+{{ end -}}
+
+{{- define "__wechat_text_list" -}}
+{{ template "rancher.title" . }}
+{{ range .Alerts.Firing }}
+{{ template "__text_single" . }}
+{{ end -}}
+{{ end -}}
+
+{{- define "slack.text" -}}
+{{ template "__slack_text_list" . }}
+{{ end -}}
+
+{{- define "__slack_text_list" -}}
+{{ range .Alerts.Firing }}
+{{ template "__text_single" . }}
+{{ end -}}
+{{ end -}}
+
+
+
+{{- define "__text_single" -}}
+{{- if eq .Labels.alert_type "event" }}
+Alert Name: {{ .Labels.alert_name}}
```

Comment:
I see so many following blocks in the template. Should we move them to the top of the `__text_single` template. 
```
Alert Name: ...
Severity: ...
Cluster Name: ...
```